src/System.Threading.Tasks.Extensions/src/System/Threading/Tasks/ValueTask.cs

```diff
@@ -0,0 +1,848 @@
+            Debug.Assert(obj == null || obj is Task || obj is IValueTaskSource);
+            return
+                obj == null ? CompletedTask :
+                obj as Task ??
+                GetTaskForValueTaskSource(Unsafe.As<IValueTaskSource>(obj));
+        }
+
+        /// <summary>Gets a <see cref="ValueTask"/> that may be used at any point in the future.</summary>
+        public ValueTask Preserve() => _obj == null ? this : new ValueTask(AsTask());
+
+        /// <summary>Creates a <see cref="Task"/> to represent the <see cref="IValueTaskSource"/>.</summary>
+        /// <remarks>
+        /// The <see cref="IValueTaskSource"/> is passed in rather than reading and casting <see cref="_obj"/>
+        /// so that the caller can pass in an object it's already validated.
+        /// </remarks>
+        private Task GetTaskForValueTaskSource(IValueTaskSource t)
+        {
+            ValueTaskSourceStatus status = t.GetStatus(_token);
+            if (status != ValueTaskSourceStatus.Pending)
+            {
+                try
+                {
+                    // Propagate any exceptions that may have occurred, then return
+                    // an already successfully completed task.
+                    t.GetResult(_token);
+                    return CompletedTask;
+
+                    // If status is Faulted or Canceled, GetResult should throw.  But
+                    // we can't guarantee every implementation will do the "right thing".
+                    // If it doesn't throw, we just treat that as success and ignore
+                    // the status.
+                }
+                catch (Exception exc)
+                {
+                    if (status == ValueTaskSourceStatus.Canceled)
+                    {
+#if !NETSTANDARD
+                        if (exc is OperationCanceledException oce)
+                        {
+                            var task = new Task<VoidTaskResult>();
+                            task.TrySetCanceled(oce.CancellationToken, oce);
+                            return task;
+                        }
+#endif
+                        return s_canceledTask;
+                    }
+                    else
+                    {
+#if NETSTANDARD
+                        var tcs = new TaskCompletionSource<bool>();
+                        tcs.TrySetException(exc);
+                        return tcs.Task;
+#else
+                        return Task.FromException(exc);
+#endif
+                    }
+                }
+            }
+
+            var m = new ValueTaskSourceAsTask(t, _token);
+            return
+#if NETSTANDARD
+                m.Task;
+#else
+                m;
+#endif
+        }
+
+        /// <summary>Type used to create a <see cref="Task"/> to represent a <see cref="IValueTaskSource"/>.</summary>
+        private sealed class ValueTaskSourceAsTask :
+#if NETSTANDARD
+            TaskCompletionSource<bool>
+#else
+            Task<VoidTaskResult>
```

Comment:
I am noticing several failures coming from these invocations involving VoidTaskResult protection level.

Reply:
Same. It's code that is not supposed to apply to netstandard. Removed.